core/src/main/rust/linker-wrapper.py

```diff
@@ -1,18 +1,44 @@
 from __future__ import absolute_import, print_function, unicode_literals
 
+import codecs
 import os
 import pipes
 import shutil
 import subprocess
 import sys
+import tempfile
+
+rustcc = os.environ['RUST_ANDROID_GRADLE_CC']
+
+if sys.platform == 'msys' or sys.platform == 'cygwin':
+    rustcc = subprocess.check_output(['/usr/bin/cygpath', '-u', rustcc], shell=False).decode('ascii').strip()
+
+args = [rustcc, os.environ['RUST_ANDROID_GRADLE_CC_LINK_ARG']] + sys.argv[1:]
+
+linkargfileName = ''
+if (sys.platform == 'msys' or sys.platform == 'cygwin') and len(''.join(args)) > 8191:
+    # response file should be use UTF-16 with BOM
+    linkargfile = tempfile.NamedTemporaryFile(delete=False)
+    linkargfile.write(codecs.BOM_UTF16_LE)
+    linkargfile.write('\n'.join(sys.argv[1:]).encode('utf-16-le'))
+    linkargfile.close()
+    linkargfileName = linkargfile.name
+    linkargfileNameW = subprocess.check_output(['/usr/bin/cygpath', '-w', linkargfileName], shell=False).decode('ascii').strip()
+    args = [rustcc, os.environ['RUST_ANDROID_GRADLE_CC_LINK_ARG'], '@' + linkargfileNameW]
 
-args = [os.environ['RUST_ANDROID_GRADLE_CC'], os.environ['RUST_ANDROID_GRADLE_CC_LINK_ARG']] + sys.argv[1:]
 
 # This only appears when the subprocess call fails, but it's helpful then.
 printable_cmd = ' '.join(pipes.quote(arg) for arg in args)
 print(printable_cmd)
 
 code = subprocess.call(args)
 if code == 0:
-    shutil.copyfile(sys.argv[sys.argv.index('-o') + 1], os.environ['RUST_ANDROID_GRADLE_TARGET'])
+    sys_argv = sys.argv
+    linkargs = list(filter(lambda s: s.startswith('@') and s.find('linker-arguments') != -1, sys.argv[1:]))
+    if linkargs != []:
+        with open(linkargs[0][1:]) as f:
+            sys_argv = f.read().splitlines()
+    shutil.copyfile(sys_argv[sys_argv.index('-o') + 1], os.environ['RUST_ANDROID_GRADLE_TARGET'])
+if linkargfileName != '':
+    os.unlink(linkargfileName)
```

Comment:
I did not test this but should this be wrapped in system check too?

Reply:
I have add system check.
if mozilla/rust-android-gradle#58 fixed, the response file will work unser msys2 and cygwin.